test/communication_test.go

```diff
@@ -0,0 +1,121 @@
+package tests
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/xlab-si/emmy/client"
+	pb "github.com/xlab-si/emmy/comm/pro"
+	"github.com/xlab-si/emmy/common"
+	"github.com/xlab-si/emmy/server"
+	"google.golang.org/grpc"
+	"log"
+	"math"
+	"math/big"
+	"net"
+	"os"
+	"testing"
+)
+
+var testGrpcServer *grpc.Server
+var testGrpcServerEndpont = "localhost:7008"
+var stopChan = make(chan bool, 1)
+
+func setupTestGrpcServer() {
+	lis, err := net.Listen("tcp", ":7008")
+	if err != nil {
+		log.Fatalf("Could not connect: %v", err)
+	}
+	testGrpcServer := grpc.NewServer(
+		grpc.MaxConcurrentStreams(math.MaxUint32),
+	)
+	pb.RegisterProtocolServer(testGrpcServer, server.NewProtocolServer())
+	go testGrpcServer.Serve(lis)
+}
+
+func teardownTestGrpcServer() {
+	go func() {
+		for {
+			select {
+			case _ = <-stopChan:
+				testGrpcServer.GracefulStop()
+			}
+
+		}
+	}()
+	stopChan <- true
```

Comment:
My bad. Actually since my last modification of `getConnection` function in `client.go` I don't think we need channels here at all. I'm slightly modifying this just to get rid of the global `testGrpcServer` variable.